### ISSUE TITLE
Dispose does not allow to re-configure the library

### ISSUE THREAD

**Issue by Xeeynamo** [DataDog/dogstatsd-csharp-client#153]
I am expecting that calling the static method `DogStatsd.Dispose()` would invalidate the last configuration sent to the static `DogStatsd.Configure`, but it still returns the exception `InvalidOperationException` with the message `Configuration for DogStatsdService already performed` when called a second time.

### Scenario 1

Does **not** throw an exception:
```
DogStatsd.Configure(configuration1);
DogStatsd.Dispose();
DogStatsd.Configure(configuration2);
```

### Scenario 2

Throws an exception:
```
DogStatsd.Configure(configuration1);
DogStatsd.Configure(configuration2);
```

## Actual behaviour

Throws an exception:
```
DogStatsd.Configure(configuration1);
DogStatsd.Dispose();
DogStatsd.Configure(configuration2);
```

**Comment by ogaca-dd** [comment]
Hi @Xeeynamo,

Sorry for my late answer.

There is an issue when running the following code:
```
DogStatsd.Configure(configuration1);
DogStatsd.Dispose();
DogStatsd.Configure(configuration2);
```

`DogStatsd.Dispose();` should throw `ObjectDisposedException`.  
The condition is wrong at https://github.com/DataDog/dogstatsd-csharp-client/blob/master/src/StatsdClient/DogStatsdService.cs#L31-L34. (It should be `if (_statsdData == null)` instead of `if (_statsdBuilder == null)`).

In your case, I think the simplest solution is to create an instance of `DogStatsdService`, dispose it and create a new one. 

Note: For an instance of `DogStatsdService`, calling `Dispose` and then `Configure` should throw  `ObjectDisposedException`. I think calling `DogStatsd.Dispose` then `DogStatsd.Configure` should have the same behavior as for `DogStatsdService`.

**Comment by Xeeynamo** [comment]
> Hi @Xeeynamo,
> 
> Sorry for my late answer.
> 
> There is an issue when running the following code:
> 
> ```
> DogStatsd.Configure(configuration1);
> DogStatsd.Dispose();
> DogStatsd.Configure(configuration2);
> ```
> 
> `DogStatsd.Dispose();` should throw `ObjectDisposedException`.
> The condition is wrong at https://github.com/DataDog/dogstatsd-csharp-client/blob/master/src/StatsdClient/DogStatsdService.cs#L31-L34. (It should be `if (_statsdData == null)` instead of `if (_statsdBuilder == null)`).
> 
> In your case, I think the simplest solution is to create an instance of `DogStatsdService`, dispose it and create a new one.
> 
> Note: For an instance of `DogStatsdService`, calling `Dispose` and then `Configure` should throw `ObjectDisposedException`. I think calling `DogStatsd.Dispose` then `DogStatsd.Configure` should have the same behavior as for `DogStatsdService`.

Hi. I addressed the issue on #154 , but honestly I prefer the approach you suggested. I was not aware it was possible to create an object that could be configured individually. Global instances are really hard to work with, isn't it? 😄

I will keep the issue open and fix the description by specifying this is happening exclusively with the static method. But be free to close if this creates noise in the list of issues.
